File: src/coag_solvers/v02_rs/src/utils/elementary_functions.rs
use std::f64::consts::PI;

pub fn gaussian(x: &Vec<f64>, mu: f64, sigma: f64) -> Vec<f64> {
    let f = |x: f64| (-(x - mu) / sigma.powf(2.)).exp() / ((2. * PI).powf(0.5) * sigma);
    x.iter().map(|x| f(*x)).collect()
}

pub fn dirac_delta(x: &Vec<f64>, i_0: usize) -> Vec<f64> {
    let f = |i: usize| match i == i_0 {
        true => 1_f64,
        false => 0_f64,
    };
    (0..x.len()).map(|i| f(i)).collect()
}

pub fn kronecker_delta(i: i32, j: i32) -> i32 {
    match i == j {
        true => 1,
        false => 0,
    }
}

File: src/coag_solvers/v01_py/src/utils/elementary_functions.py
import numpy as np
from numba import jit


@jit(nopython=True)
def gaussian(x, mu, sigma):
    y = np.exp(-((x-mu)/sigma)**2) / ((2*np.pi)**.5 * sigma)
    return y


@jit(nopython=True)
def dirac_delta(x, i_x0):
    y = np.zeros(len(x))
    y[i_x0] = 1
    return y


@jit(nopython=True)
def kronecker_delta(i, j):
    return 1 if i == j else 0


@jit(nopython=True)
def heaviside_theta(x):
    if x < 0:
        return 0
    elif x == 0:
        return 1/2
    else:
        return 1
